homeassistant/components/energy/types.py

```diff
@@ -0,0 +1,27 @@
+"""Types for the energy platform."""
+from __future__ import annotations
+
+from typing import Awaitable, Callable, TypedDict
+
+from homeassistant.core import HomeAssistant
+
+
+class SolarForecastType(TypedDict):
+    """Return value for solar forecast."""
+
+    wh_hours: dict[str, float | int]
+
+
+GetSolarForecastType = Callable[
+    [HomeAssistant, str], Awaitable["SolarForecastType | None"]
```

Comment:
Do we need the quotes?
```suggestion
    [HomeAssistant, str], Awaitable[SolarForecastType | None]
```

Reply:
Yes, because it's not inside a type annotation but is stored as a variable, so new rules apply. Python 3.10+ should be fine.